### PR TITLE
Lavaland bomb runtime fix

### LINKED ISSUE #35158: lava fire armor runtime
https://tgstation13.org/parsed-logs/basil/data/logs/2018/01/30/

```
The following runtime has occurred 819 time(s).
runtime error: Cannot read null.fire
proc name: burn stuff (/turf/open/lava/proc/burn_stuff)
  source file: lava.dm,103
  usr: null
  src: the lava (62,45,5) (/turf/open/lava/smooth/lava_land_surface)
```
```
[21:36:28]proc name: burn stuff (/turf/open/lava/proc/burn_stuff)
[21:36:28]  source file: lava.dm,103
[21:36:28]  usr: null
[21:36:28]  src: the lava (62,45,5) (/turf/open/lava/smooth/lava_land_surface)
[21:36:28]  call stack:
[21:36:28]the lava (62,45,5) (/turf/open/lava/smooth/lava_land_surface): burn stuff(null)
[21:36:28]the lava (62,45,5) (/turf/open/lava/smooth/lava_land_surface): process(20)
[21:36:28]Objects (/datum/controller/subsystem/obj): fire(0)
[21:36:28]Objects (/datum/controller/subsystem/obj): ignite(0)
[21:36:28]Master (/datum/controller/master): RunQueue()
[21:36:28]Master (/datum/controller/master): Loop()
[21:36:28]Master (/datum/controller/master): StartProcessing(0)
```

https://github.com/tgstation/tgstation/blob/956d4ed81d1a747bd9596cabc1a4c2cb023b76bb/code/game/turfs/simulated/lava.dm#L103

#34793 @vuonojenmustaturska

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Fixes #35158

~This seems to be caused by yet-to-be-initialized ores etc getting thrown into/across lava. The nearby code kind of ought to be refactored but that's a job for another PR, currently touching lava permanently alters objects by removing their fireproofness, making them flammable and changing their armor.~

Turns out cleanables were the culprit, they weren't getting initialized because of the changes in #35072 

I also took the opportunity to fix some longstanding issues with their initialization (decal replacing now works by initializing decals deleting themselves in favor of the existing one, instead of an existing decal being deleted, sometimes before its initialization has run)

### CODE REVIEW

File: code/game/turfs/simulated/lava.dm [hunk 1]
```diff
@@ -100,7 +100,7 @@
 				O.resistance_flags |= FLAMMABLE //Even fireproof things burn up in lava
 			if(O.resistance_flags & FIRE_PROOF)
 				O.resistance_flags &= ~FIRE_PROOF
-			if(O.armor.fire > 50) //obj with 100% fire armor still get slowly burned away.
+			if(istype(O.armor, /datum/armor) && O.armor.fire > 50)
```
Comment (DaedalusGame) [inline code/game/turfs/simulated/lava.dm]: can this be just istype(O.armor) ?

Comment (Cyberboss) [reply code/game/turfs/simulated/lava.dm]: yes

**Comment by Cyberboss** [comment]
If the game is running, at no point should anything not be initialized after the line that `new`'d it finishes, so I'm not convinced this is a good fix

**Comment by Cyberboss** [comment]
An alternative explanation is something is sleeping in New/Initialize which is a big nono

**Comment by DaedalusGame** [comment]
This can't be right. The stacktrace in the issue says `burn_stuff(null)`. There doesn't appear to be proper handling for passing null into it.

**Comment by Cyberboss** [comment]
`burn_stuff(null)` just means burn everything on the turf

**Comment by vuonojenmustaturska** [comment]
yeah this is actually related to a recent change that i merged 🥇

**Pull request merged**